### PR TITLE
Add timezone setting

### PR DESCRIPTION
**Pull request by LinuxSuRen**
export TZ as an env variable

### CODE REVIEW

**Comment by batmat** [comment]
Looks interesting, but could you please see to add tests? Thanks!

**Comment by LinuxSuRen** [comment]
Do we have some documents about how to add tests? The manual test is ok.

**Comment by oleg-nenashev** [comment]
Thanks @LinuxSuRen !
Apart from tests (which are optional IMHO), some documentation is needed before it is merged

File: Dockerfile [hunk 1]
```diff
@@ -62,6 +62,10 @@ ENV JENKINS_UC_EXPERIMENTAL=https://updates.jenkins.io/experimental
 ENV JENKINS_INCREMENTALS_REPO_MIRROR=https://repo.jenkins-ci.org/incrementals
 RUN chown -R ${user} "$JENKINS_HOME" /usr/share/jenkins/ref
 
+# setting timezone
+ENV TZ=UTC
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
```
Comment (batmat) [inline Dockerfile]: Also, I tend to think this is not how we should do this: using a `RUN` step, this means one has to rebuild the whole image to be able to adapt the TZ.

What does this PR provide that cannot be provided mapping the TZ file when executing `docker run`?

```
date ; \
docker run jenkins/jenkins:lts date ; \
docker run -v /usr/share/zoneinfo/CET:/etc/localtime jenkins/jenkins:lts date
Thu Dec 20 14:52:37 CET 2018
Thu Dec 20 13:52:37 UTC 2018
Thu Dec 20 14:52:38 CET 2018
```

Comment (vbricard) [reply Dockerfile]: +1
I agree but I rather map the server's file /etc/localtime.

Comment (batmat) [reply Dockerfile]: I don't think it's in scope here, given we do not provide native Docker images for Windows, at least yet.

**Comment by batmat** [comment]
Going to close this given the discussion we had above.

@LinuxSuRen please feel free to come and explain your use case, so we can reopen/consider this, but I really feel like the right path to handle timezones is https://github.com/jenkinsci/docker/pull/777#discussion_r243493541

Having to rebuild the image to change the timezone seems much more complex than just mapping the right timezone file, which I think is also the correct way to do that in general, even apart from the context of the Jenkins Docker image.

Thanks a lot for the contribution @LinuxSuRen, please keep them coming! :heart:

**Comment by LinuxSuRen** [comment]
@batmat Yes, I just wish I could find a better solution to setting the timezone. The mapping way still is complex.

**Comment by batmat** [comment]
I tend to disagree. At least, I could agree it's possibly a _bit_ complex but still way less than rebuilding the whole image to change it :)

**Comment by LinuxSuRen** [comment]
I agree with that the rebuilding is bad. My expect way is that `docker run jenkins/jenkins -e TZ=UTC`.

**Pull request closed without merging**